src/device.rs

```diff
@@ -110,6 +110,32 @@ impl<'a> Device<'a> {
         }
     }
 
+    /// Returns the first parent with a given subsystem and devtype
+    pub fn parent_with_subsystem_devtype<T: AsRef<OsStr>, U: AsRef<OsStr>>(&self, subsystem: T, devtype: Option<U>) -> ::Result<Option<Device>> {
```

Comment:
Curious, if a user types `device.parent_with_subsystem_devtype("...", None)`, will Rust be able to infer a type for `U`?